### ISSUE TITLE
Annotation: hasDbXref option?

### ISSUE THREAD

**Issue by cmrn-rhi** [obi-ontology/obi#1576]
Noticed that OBI does not to use the `hasDBXref` annotation [http://www.geneontology.org/formats/oboInOwl#hasDbXref] - what annotation should be used for database cross-references?

**Comment by zhengj2007** [comment]
OBI used [IAO_0000119](http://purl.obolibrary.org/obo/IAO_0000119):definition source to reference the terms in other resources.

Generally OBI does not provide the mapped terms in other resources in the ontology. When OBI wants to use a term in an existing ontology, OBI will import the term if it is in the active OBO Foundry ontology. Otherwise, OBI will create a new term in the OBI and use IAO_0000119:definition source to indicate the source of definition of the term.

The annotation of hasDBXref is not used consistently in the OBO Foundry ontologies that why it is not used in the OBI.

Here are some discussions related to your question:
https://github.com/dcppc/metadata-matrix/issues/2
https://github.com/information-artifact-ontology/ontology-metadata/issues/31

**Comment by ddooley** [comment]
The GenEpiO group circled back to this today in discussion.  We have a plan to mint lots of GenEpiO terms, then gradually have them migrate to other ontologies like OBI.  For our end-users we'd like to enable an OLS search on a deprecated GenEpiO id to directly yield its updated OBI term after the deprecation occurs.  We assume we can't have OBI add hasDbXref pointing to the deprecated GenEpiO term ...

A solution:? Is it ok instead to have GenEpiO annotate an imported OBI term with hasDbXref pointing to deprecated GenEpiO term?  That way the GenEpiO term could be found in OLS or OntoBee without users having to have the "search obsolete terms" option turned on (and getting the obsolete term result and its "term replaced by" link leading to new term.)

**Comment by jamesaoverton** [comment]
@ddooley So you plan to use IAO ['term replaced by'](http://purl.obolibrary.org/obo/IAO_0100001) to link from your deprecated GenEpiO term to the replacement OBI term, but you think that will not be enough for OLS search?

**Comment by ddooley** [comment]
Yes, we'll use IAO term replaced by in the GenEpiO term.  But say GenEpiO deprecates a term with label X in favour of OBI term label Y.   Now when we search for X in OLS, we find nothing unless we turn on "search deprecated terms".  Rather than having to do that, what would be neat is if we actually got a hit on GenEpiO's use of Y, because Y had a hasDBXref annotation in GenEpiO that specifies X.

**Comment by cmrn-rhi** [comment]
This is a user experience thing, as for many users who are not ontologists/curators - if they don't find a term they are looking for, the generally do not know or recall to turn on the deprecated/obsolete term search option.

**Comment by jamesaoverton** [comment]
You have things you need to accomplish, and I see how this would help you get closer. I won't stand in your way. Since you seem to be asking for opinions, my opinions are:

1. it would be better to just create the terms in OBI in the first place
2. it just seems wrong to me to change ontology content because of the behaviour of one ontology browser
3. I don't see why the inverse of 'term replaced by' is 'hasDBXref'
4. there's a push in OBO *not* to mess with imported terms, even by adding harmless annotations

**Comment by cmrn-rhi** [comment]
1. That is what we have tried, but due to long processing time we end up in situations where the controlled vocabulary is needed before they can be processed in the target domain. We're trying to figure out a workflow to deal with terms that take years to process (or get rejected and try again elsewhere).
2. That's fair, we can try to get more users onto ontobee as at least when there are no results the "show deprecated terms" option is more obvious. We can try to direct users to it more, and get them to follow the "term replaced by" annotation. We have been and still are working on educating people on this process, but it can be overwhelming at times and be seen as a rationale against going the ontology route.
3. It's not logically the inverse, more it just seemed like and accessible annotation for associating a term with a previously assigned ontology identifier that exists in resources outside of the ontology it has been deprecated in.
4. I don't think I'm experienced enough in the OBO community to respond to this discussion, but I do think there are reasons to do so within an application ontology on the user-experience side of things - whether that is justified I'm sure is already under debate.

**Comment by cmrn-rhi** [comment]
4. Follow-up: and just to clarify, I mean only _within_ the application ontology with that comment.

**Comment by cmrn-rhi** [comment]
So considering OBI uses "definition source" and avoids the ambiguous "hasDBXref", perhaps a better question would be: when requesting to "rehome" a term - would it be feasible to have a "definition source" annotation that references the previous ID or 
PURL that has been deprecated as a consequence of the "rehoming"?

**Comment by jamesaoverton** [comment]
'definition source' does seem better to me than 'hasDBXref'.
